client/packages/system/src/Item/context/useUnitVariant.tsx

```diff
@@ -1,71 +1,79 @@
-import React, { PropsWithChildren, createContext, useContext } from 'react';
-import { useItemVariants } from '../api';
+import { useUnitVariantList } from '../api';
+import { create } from 'zustand';
+import isEqual from 'lodash/isEqual';
+import { UnitVariantNode } from '@common/types';
 import { NumUtils } from '@common/utils';
+import { useEffect } from 'react';
 
-export const useUnitVariant = (itemId?: string) => {
-  const { data } = useItemVariants();
-  const item = data?.find(item => item.id === itemId);
-
-  const asPackUnit = (packSize: number) => {
-    return (
-      item?.variants.find(variant => variant.packSize === packSize)
-        ?.shortName || `${item?.unit} x ${packSize}`
-    );
-  };
-
-  // calculates SOH, AMC, MOS, Target, Suggested and Requested based on default or user selected variant
-  const numberOfPacksFromQuantity = (
-    totalQuantity: number,
-    packSize: number
-  ) => {
-    return NumUtils.round(totalQuantity / packSize, 2);
-  };
-
-  const options = item?.variants ?? [];
-
-  // const quantityFromNumberOfPacks - reverse of above
-
-  // set in app data
-  const setDefaultOption = () => {};
-
-  const defaultOption = () => {
-    // TODO: search app data first to see if user has preferred preference
-    return item?.variants.find(variant =>
-      variant.longName.includes(item?.unit)
-    );
+interface UnitState {
+  // From back end
+  items: {
```

Comment:
there is a little bit of mapping from array to object, makes it easier to select that bit of state